### DISCUSSION TITLE
macos window title not changing with escape sequence

### DISCUSSION THREAD

**Discussion by mightybs** [ghostty-org/ghostty#4113] [Feature Requests, Ideas] [answered]
The [documentation](https://ghostty.org/docs/config/reference#title) suggests that escape sequences to change the title will work, but running this does not affect title change like it does in other terminals.

```
printf '\033]0;foo\007'
```

I use this to better organize open windows, allowing different apps to set titles, or myself set a title when wanted, but by default I use this to name my Windows by the date that they were opened to help find stuff I've recently opened and didn't further organize.

```
title="$(date +'=== %Y-%m-%d - %H:%M:%S - %A (Week:%V Day:%j) ===';)"
printf "\033]0;$title\007"
```

**Comment by mightybs** [comment]
I just decided to try this from bash, rather than zsh, and it's working from bash.

**Reply by mightybs** [reply]
If I start a new window and try to change the title it will not work, but entering another shell, such as zsh or bash and then running the escape sequence, the title change will work.

**Reply by mightybs** [reply]
Furthermore, once you exit that new shell, then the changed title is reverted back to what it was before launching that new shell, and we're back to the escape sequences not working.

**Comment by francof** [comment]
I can confirm this still happens, works perfectly if starting a new shell (even if it's the same shell, starting zsh inside zsh), but doesn't work in the main one, not even sourcing ~/.zshrc

**Reply by rhodes-b** [reply, marked as answer]
thats because the very first shell has automatic shell-integration which on its own has title reporting after commands are invoked

you can either disable shell-integration entirely or `shell-integration-features=no-title`

**Reply by francof** [reply]
perfect, did that and now can manage the title by myself, thanks!
